code-challenges.test.js

```diff
@@ -13,38 +15,158 @@
+// need to pass uppercased and rejoined split array thing back into original array or new array i dunno
+
+// const firstCap = (array) => {
+//     let splitArr = array.map(arrayElement => {
+//        let splitElement = arrayElement.split("")
+//        return splitElement[0].toUpper()
+//     })
+//     return splitArr.join("")
+// }
+
+//not working so lets try something new
+//create a function that splits an array of strings into single arrays
+//line 99 correctly uppercases first word
+// figure out how to get it to take in an array and you are good
+
+// const splitArrIntoMoreArrays = (array) => {
+//     let newWord = array.split()
+//     for (let i = 0; i < array.length; i++) {
+//         const element = array[i].split();
+//         return element.charAt(0).toUpperCase() + element.slice(1)
+
+
+//     }
+//     return array
+// }
+
+//code below can uppercase the first letter in a string array
+//takes 
+//const newWord = randomNouns1[0].charAt(0).toUpperCase() + randomNouns1[0].slice(1) 
+
+//i just learned that i could not print out the console log randomNoun2 or randomNoun1 because the function call was before the variable declaration
+
+// console.log(randomNouns2);
+
 // a) Create a test with expect statements for each of the variables provided.
 
 var randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
 // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
+
 var randomNouns2 = ["temperature", "database", "chopsticks", "mango", "deduction"]
 // Expected output: ["Temperature", "Database", "Chopsticks", "Mango", "Deduction"]
 
 
+describe("firstUpper", () => {
+
+    // describes that the function does.
+    it("uppercase the first letter of each word in the array", () => {
+  
+      //an expect method, nested within the test block, calling on the divByThree() function, followed by the .toEqual() matcher that checks the expected output of the function return.
+      //call function (firstUpper) with arguments randomNouns1 and randomNouns2 
+      //verify that the output is as expected
+      expect(firstUpper(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+      expect(firstUpper(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango", "Deduction"])
+      
+    })
+})
 
 // b) Create the function that makes the test pass.
+// create function called firstUpper, with parameter array (randomNouns1, 2)
+// using map function, cycle over each word and feed it into value
+// for each value, set the first character (.charAt(0)) to an uppercase value (.toUpperCase()) and return value
+//join by string concatination, the upper cased value + the rest of the word (value.slice(1))
+// then return that to the new variable created (newWord)
+
+const firstUpper = (array) => {
+    let newWord = array.map(value => {
+        return value.charAt(0).toUpperCase() + value.slice(1)
+    })
+    return newWord
+}
+
```

Comment:
Nice job! Just a small note - if you are creating a variable that's only purpose is to get returned, you can often skip the variable declaration and just return your statement.